Task8/README.md

```diff
@@ -0,0 +1,158 @@
+
+#EXAMPLE
+gcloud iam service-accounts create zenml-sa
+```
+Grant roles to the service account
+```
+gcloud projects add-iam-policy-binding <PROJECT_ID> --member="serviceAccount:<SA-NAME>@<PROJECT_ID>.iam.gserviceaccount.com" --role=<ROLE>
+
+#EXAMPLE
+gcloud projects add-iam-policy-binding zenml-vertex-ai --member="serviceAccount:zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com" --role=roles/storage.admin
+gcloud projects add-iam-policy-binding zenml-vertex-ai --member="serviceAccount:zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com" --role=roles/aiplatform.admin
+```
+Generate the key file (it generates in the same location)
+```
+gcloud iam service-accounts keys create <FILE-NAME>.json --iam-account=<SA-NAME>@<PROJECT_ID>.iam.gserviceaccount.com
+
+EXAMPLE
+gcloud iam service-accounts keys create zenml-sa-file.json --iam-account=zenml-sa@zenml-vertex-ai.iam.gserviceaccount.com
+```
+Set the environment variable:
+
+To use service accounts with the Google Cloud CLI, you need to set an environment variable where your code runs
+```
+export GOOGLE_APPLICATION_CREDENTIALS=<KEY-FILE-LOCATION>
+
+#EXAMPLE
+export GOOGLE_APPLICATION_CREDENTIALS="/Users/co2yd4bcjgh6/Library/CloudStorage/OneDrive-Personal/myWorkspace/github/DataScience/Task8/zenml-sa-file.json"
+```
+1.3. Create a `GCP bucket` to which Vertex should output any artifacts from the training run
+```
+gsutil mb -l <REGION> gs://bucket-name
+
+#EXAMPLE
+gsutil mb -l europe-west1 gs://zenml-bucket
+```
+
+1.4. Configure and enable container registry in GCP
+
+This registry will be used by ZenML to push your job images that Vertex will use.
+
+a) [Enable](https://cloud.google.com/container-registry/docs) Container Registry
+b) [Authenticate](https://cloud.google.com/container-registry/docs/advanced-authentication) your local `docker` cli with your GCP container registry 
+
+```
+docker pull busybox
+docker tag busybox gcr.io/<PROJECT-ID/busybox
+docker push gcr.io/<PROJECT-ID>/busybox
+```
+You can view this image in the gcp console (container registry) or by running
+```
+gcloud container images list-tags gcr.io/<PROJECT-ID>/busybox
+```
+
+1.5. [Enable](https://console.cloud.google.com/marketplace/product/google/aiplatform.googleapis.com?q=search&referrer=search&project=cloudguru-test-project) `Vertex AI API`
+
+### 2. Set up the components requiered for your stack
+2.1. Install integrations
+```
+zenml integration install sklearn vertex gcp
+
+# Initialize
+zenml init
+```
+2.2. Set the bucket created earlier
+```
+zenml artifact-store register <NAME> --type=gcp --path=<GCS_BUCKET_PATH>
+
+# EXAMPLE
+zenml artifact-store register gcp-store --type=gcp --path=gs://zenml-bucket
+```
+2.3. Create the vertex step operator
+```
+zenml step-operator register <NAME> \
+    --type=vertex \
+    --project=<PROJECT-ID> \
+    --region=<REGION> \
+    --machine_type=<MACHINE-TYPE> \
+    --base_image=<CUSTOM_BASE_IMAGE> #this can be left out if you wish to use zenml's default image
```

Comment:
Can you only have one base image for all steps?

Reply:
I verified it on their slack channel and yes, you can have only one step operator (with one base image) per stack

Reply:
> hi @halvgaard, I started writing the kaggle notebook but discovered some bugs and errors along the way. So I asked on slack, please have a look at this [slack thread](https://zenml.slack.com/archives/C01FWQ5D0TT/p1652286532072619). What do you think? Should I do just the presentation?

Check the response in slack. Maybe put it one of the places he suggests. I think the zenfiles repo has pretty high standards so would be cool and very visible.